Connection.php

```diff
@@ -35,19 +35,37 @@ class Connection extends Component
      */
     public $autodetectCluster = true;
     /**
-     * @var array cluster nodes
+     * @var array[] cluster nodes
      * This is populated with the result of a cluster nodes request when [[autodetectCluster]] is true.
+     * Additional special options:
+     *  - `auth`: overrides [[auth]] property. For example:
+     * ```php
+     * [
+     *  'http_address' => 'inet[/127.0.0.1:9200]',
+     *  'auth' => ['username' => 'yiiuser', 'password' => 'yiipw'], // Overrides the `auth` property of the class with specific login and password
+     *  //'auth' => ['username' => 'yiiuser', 'password' => 'yiipw'], // Disabled auth regardless of `auth` property of the class
+     * ]
+     * ```
      * @see http://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-info.html#cluster-nodes-info
      */
     public $nodes = [
         ['http_address' => 'inet[/127.0.0.1:9200]'],
     ];
     /**
-     * @var array the active node. key of [[nodes]]. Will be randomly selected on [[open()]].
+     * @var string the active node. Key of one of the [[nodes]]. Will be randomly selected on [[open()]].
      */
     public $activeNode;
-    // TODO http://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_configuration.html#_host_configuration
+
+    /**
+     * @var array|\Closure Authorization data used to connect to the ElasticSearch node.
+     * Possible array elements:
+     *  - `username`: the username for authentication.
+     *  - `password`: the password for authentication.
+     * Array MUST contain both `username` and `password`, when is set.
```

Comment:
I guess it should be `Array either MUST contain both username or password on not contain any authentication credentials.`